### ISSUE TITLE
Remove EventListenerGuard

### ISSUE THREAD

**Issue by jasonmalinowski** [dotnet/roslyn#25822]
In 958ca00a68738ac61f5c8b0acce7d140567d2243 @jaredpar added a workaround for a framework bug. That bug has since shipped in 4.6.2, and so we can probably look at removing it.

**Comment by jaredpar** [comment]
I don't think we can blanket remove this yet. There are still plenty of machines running 4.6.1 out there. At best we can conditionally remove it. Probably can't fully remove it until we're willing to make our unit tests projects target net462.
